cli/src/diff_util.rs

```diff
@@ -2052,56 +2063,89 @@ pub fn show_diff_stats(
 
     // Entries format like:
     //   path/to/file | 123 ++--
+    // or, for binary files:
+    //   path/to/file | (binary) +1234 bytes
     //
     // Depending on display widths, we can elide part of the path,
     // and the the ++-- bar will adjust its scale to fill the rest.
 
     let max_path_width = ui_paths.iter().map(|s| s.width()).max().unwrap_or(0);
-    let max_diffs = stats
-        .entries()
-        .iter()
-        .map(|stat| stat.added + stat.removed)
-        .max()
-        .unwrap_or(0);
-
-    let number_width = max_diffs.to_string().len();
-    // Available width to distribute between path and bar:
-    let available_width = display_width.saturating_sub(" | ".len() + number_width + " ".len());
-    // Always use at least a tiny bit of room
-    let available_width = max(available_width, 5);
-
-    let max_path_width = max_path_width.min((0.7 * available_width as f64) as usize);
-    let max_bar_length = available_width.saturating_sub(max_path_width);
+
+    // Compute the largest number that will be displayed for both line diffs and
+    // binary files, to measure its width.
+    let mut max_diffs = 0;
+    let mut max_bytes = 0;
+    for stat in stats.entries() {
+        if let Some((added, removed)) = stat.added_removed {
+            max_diffs = max_diffs.max(added + removed);
+        } else {
+            max_bytes = max_bytes.max(stat.bytes_delta.abs());
+        }
+    }
+
+    let max_binary_output_width = if max_bytes > 0 {
+        "(binary) +".len() + max_bytes.to_string().len() + " bytes".len()
+    } else {
+        0
+    };
+
+    // Choose how many columns to use for the path.  The bar will use the rest.
+    // Always assume at least a tiny bit of room:
+    let available_width = max(display_width.saturating_sub(" | ".len()), 5);
+    let max_path_width = max_path_width
+        .min(available_width.saturating_sub(max_binary_output_width))
```

Comment:
nit: maybe we should subtract `diff_number_width + margin` as well for consistency?

Reply:
I think after writing this comment, I think I convinced myself I should just write the code, since it would be shorter than this comment anyway.  I will update this patch with the full fix.